static/js/containers/DashboardPage.js

```diff
@@ -229,6 +237,36 @@ class DashboardPage extends React.Component {
     }
   };
 
+  handleCoupon = () => {
+    const { coupons, dispatch, location: { query } } = this.props;
+
+    if (!query.coupon) {
+      // If there's no coupon code in the URL query parameters,
+      // there's nothing to do.
+      return;
+    }
+
+    if ( coupons.fetchPostStatus !== undefined ) {
+      // If we've already launched a coupon fetch, don't launch another one
+      return;
+    }
+
```

Comment:
This function isn't dependent on the value of `state.coupons.coupons`, so it doesn't matter if it executes before or after that GET request returns. I suppose that you could have a race condition like this:

1. launch first `fetchCoupons()` API request
2. launch `attachCoupon()` API request
3. `attachCoupon()` returns, launch second `fetchCoupons()` API request
4. second `fetchCoupons()` returns, updates Redux store with accurate information
5. first `fetchCoupons()` _finally_ returns, updates Redux store with stale information

If that's the race condition you're thinking of, wouldn't it make more sense to abort the first `fetchCoupons()` API request in step 3, instead of waiting for it to finish?